llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -5559,14 +5578,25 @@ InstructionCost LoopVectorizationCostModel::computePredInstDiscount(
 InstructionCost LoopVectorizationCostModel::expectedCost(ElementCount VF) {
   InstructionCost Cost;
 
+  // If the vector loop gets executed exactly once with the given VF, ignore the
+  // costs of comparison and induction instructions, as they'll get simplified
+  // away.
+  SmallPtrSet<Instruction *, 2> ValuesToIgnoreForVF;
+  auto TC = PSE.getSE()->getSmallConstantTripCount(TheLoop);
+  if (VF.isFixed() && TC == VF.getFixedValue()) {
+    assert(!foldTailByMasking());
```

Comment:
Was this intentionally added? IIUC this is causing crashes as per the earlier comment thread (which seems to have been dropped by Github here)

Reply:
Oops, no. I was checking the behaviour and forgot to undo the changes